assets/javascripts/overview.js

```diff
@@ -287,8 +287,32 @@ function showAddCommentsDialog() {
   modal.show();
 }
 
-function addComments(form) {
+function restartJobsWithComment(btn) {
+  const form = btn.form;
   const text = form.elements.text.value;
+  const jobs = btn.dataset.jobs
+    .split(',')
+    .map(Number)
+    .filter(n => !isNaN(n));
+  const apiUrl = btn.dataset.url;
+  if (!text.length) {
+    return window.alert("The comment text mustn't be empty.");
+  }
+
+  const progressIndication = document.getElementById('add-comments-progress-indication');
+  const controls = document.getElementById('add-comments-controls');
+  progressIndication.style.display = 'flex';
+  controls.style.display = 'none';
+  restartJob(apiUrl, jobs, text);
+  progressIndication.style.display = 'none';
+  controls.style.display = 'inline';
+  window.addCommentsModal.hide();
```

Comment:
How is that going to work? Have you tested this at all when restarting takes a notable amount of time (which it sometimes will in production)?

The problem I see here is that `restartJob()` will return immediately and not only when the jobs have been restarted. So the subsequent hiding of the progress bar and modal will also happen immediately. *Maybe* the following would work:

```suggestion
  restartJob(apiUrl, jobs, text).finally(() => {
    progressIndication.style.display = 'none';
    controls.style.display = 'inline';
    window.addCommentsModal.hide();
  });
```

You can test whether this works locally by putting a sleep in the web UI controller code for restarting jobs. The progress bar should then stay visible at least as long as the sleep takes.

Reply:
I see that you applied my suggested change. Have you also tested whether it works?

Reply:
yes, as i mentioned on the standup.